### DISCUSSION TITLE
Dinamically add a calendar via socket notification

### DISCUSSION THREAD

**Discussion by SimoneMSR** [MMRIZE/MMM-CalendarExt3#177] [Q&A]
I would like to know if it is possible to dinamically add and show a calendar, perhaps using the command suggested in the `calendar` module docs. Up to now, I am not able to do this from a node_helper.

`this.io.of("calendar").emit("ADD_CALENDAR",{url : "...."});`

**Comment by eouia** [comment]
The question you asked isn't directly related to this "MMM-Calendar3Ext module." 
This module only receives event data sent from the "default calendar" module and displays it on the screen. Therefore, any additions to the calendar need to be handled within the "default calendar" module.

Let's take a closer look at your question. 
You mentioned you want to add a calendar "dynamically," but what exactly do you mean by that? Under what conditions should the calendar be added dynamically? (e.g., should the calendar be added at a specific time, or when a particular hardware button is pressed?) Why does the calendar need to be added dynamically in the first place? For example, if you need to show two types of calendars, A and B, and under certain conditions, show three types, A, B, and C, then you could initially load calendars A, B, and C, but only display A and B until the condition is met to display events from calendar C. How would this differ from your intended approach? The only situation where you would need to add a calendar dynamically is if the URL of the calendar to be loaded is only known at runtime. Is this the case?

**Reply by SimoneMSR** [reply]
I imagined what you see, and I tought to write the question also the the main MagicMirror forum. Thanks for your indication, though.

My situation is the last one you imagined: I know the calendar url only at runtime.

**Reply by eouia** [reply]
As I wrote before, that relies on the default calendar module.
